### PR TITLE
reward is recieved based on application status

### PR DESCRIPTION
**Pull request by dhruvja**

### CODE REVIEW

File: programs/candidate_staking/src/lib.rs [hunk 1]
```diff
@@ -52,6 +52,8 @@ pub mod candidate_staking {
                 msg!("You can transfer");
                 msg!("Transfer is initiated");
 
+                ctx.accounts.base_account.staked_amount = amount;
```
Comment (blasrodri) [inline programs/candidate_staking/src/lib.rs:55]:
```
ctx.accounts.base_account.staked_amount += amount
```

Comment (dhruvja) [reply programs/candidate_staking/src/lib.rs:55]: ok will change that

Comment (blasrodri) [reply programs/candidate_staking/src/lib.rs:55]: check also for overflow

Comment (dhruvja) [reply programs/candidate_staking/src/lib.rs:55]: we are checking the overflow in the condition before

Comment (dhruvja) [reply programs/candidate_staking/src/lib.rs:55]: we are checking if is less than the max staked amount

Comment (blasrodri) [reply programs/candidate_staking/src/lib.rs:55]: ok, well done

**Pull request merged**